src/extensions/web/Microsoft.DotNet.UpgradeAssistant.Extensions.Web/WebExtension.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.DotNet.UpgradeAssistant.Dependencies;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Web
+{
+    public class WebExtension : IExtensionServiceProvider
+    {
+        public void AddServices(IExtensionServiceCollection services)
+        {
+            if (services is null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            services.Services.AddTransient<IDependencyAnalyzer, NewtonsoftReferenceAnalyzer>();
+            services.Services.AddScoped<IUpdater<ConfigFile>, WebNamespaceConfigUpdater>();
```

Comment:
I know you didn't change this, but the Transient/Scoped difference stands up when these are side-by-side here :) I don't think there's any reason the config updater needs to have a scoped lifetime. We may want to make it transient for consistency.

Reply:
Good call. I've created an issue to address the bigger picture here of how we decide to scope things.